src/app/util/attribute-storage.cpp

```diff
@@ -265,7 +265,7 @@ uint16_t emberAfGetDynamicIndexFromEndpoint(EndpointId id)
 
 CHIP_ERROR emberAfSetDynamicEndpoint(uint16_t index, EndpointId id, const EmberAfEndpointType * ep,
                                      const Span<DataVersion> & dataVersionStorage, Span<const EmberAfDeviceType> deviceTypeList,
-                                     EndpointId parentEndpointId)
+                                     chip::CharSpan endpointUniqueId, EndpointId parentEndpointId)
```

Comment:
A few things:

1.  This is an API change, which is why you had to change a whole bunch of other code.  What we should probably do is keep both APIs (i.e. add a second function signature that has the unique ID, internally make the version that does not take that call the one that does an pass an empty CharSpan.
2. Please take out the unnecessary `chip::` prefixes throughout.